include/zephyr/mgmt/mcumgr/smp/smp_client.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_SMP_CLIENT_
+#define H_SMP_CLIENT_
+
+#include <zephyr/net/buf.h>
+#include <mgmt/mcumgr/transport/smp_internal.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief SMP client send response callback function
+ *
+ * @param nb net_buf for response
+ *
+ * @param user_data same user data returned than for request
+ *
+ * @return 0 on success, #mcumgr_err_t code on failure.
+ */
+typedef int (*smp_client_res_fn)(struct net_buf *nb, void *user_data);
+
+
+/**
+ * @brief SMP client response handler
+ *
+ * @param nb response net_buf
+ *
+ * @param res_hdr Parsed SMP header
+ *
+ * @return 0 on success, #mcumgr_err_t code on failure.
+ */
+int smp_client_single_response(struct net_buf *nb, const struct smp_hdr *res_hdr);
+
+/**
+ * @brief SMP net buf allocate and SMP header pre-write
+ *
+ * @param group SMP grouo id
+ *
+ * @param command_id Parsed SMP header
+ *
+ * @param op SMP operation type
+ *
+ * @return      A newly-allocated buffer net_buf on success;
+ *              NULL on failure.
+ */
+struct net_buf *smp_client_buf_allocation(uint16_t group, uint8_t command_id,
+					  uint8_t op);
+/**
+ * @brief Frees an SMP client net_buf
+ *
+ * @param nb    The net_buf to free.
+ */
+void smp_client_buf_free(struct net_buf *nb);
+
+
+/**
+ * @brief SMP client data send request
+ *
+ * @param nb net_buf builded SMP packet for send
+ *
+ * @param cb Callback for response handler
+ *
+ * @param user_data user defined data pointer which will be returned back to response callback
+ *
+ * @param timeout_in_sec Define How many times re-sended with CONFIG_SMP_CMD_RETRY_TIME period.
```

Comment:
this comment does not make sense, it's the timeout in seconds, but the comment suggests it's actually not in seconds and is actually the number of times the retry is sent?